libs/accounts/recovery-phone/src/lib/recovery-phone.service.ts

```diff
@@ -80,17 +80,55 @@ export class RecoveryPhoneService {
     phoneNumber: string,
     getFormattedMessages: (code: string) => Promise<FormattedMessages>
   ) {
+    // Check if the service has been disabled.
     if (!this.config.enabled) {
       throw new RecoveryPhoneNotEnabled();
     }
 
-    if (this.config.sms && this.config.sms.validNumberPrefixes) {
+    console.log(
+      '!!! setup phone number 2',
+      this.config.sms?.validNumberPrefixes,
+      phoneNumber
+    );
+
+    // Reject numbers that do not match our set of sanctioned prefixes.
+    if (this.config.sms?.validNumberPrefixes) {
       const allowed = this.config.sms.validNumberPrefixes.some((check) => {
         return phoneNumber.startsWith(check);
       });
 
       if (!allowed) {
+        this.metrics.increment('number_prefix.blocked');
         throw new RecoveryNumberNotSupportedError(phoneNumber);
+      } else {
+        this.metrics.increment('number_prefix.allowed');
+      }
+    }
+
+    // Call Twilio to get some info about the number, and check if it resides in
+    // a country that we support sending sms to.
+    if (this.config.sms?.validCountryCodes) {
+      const lookupData = await (async () => {
+        try {
+          return await this.smsManager.phoneNumberLookup(phoneNumber, '');
```

Comment:
IIUC this is an additional call to Twilio...?  I remember that there was a PR where lookup data(?) was being cached.  So we are not really making an extra call to Twilio per phone number for this?

Reply:
There is another call to this api, but it happens a later point in time when the user actually confirms the account. At this point in time we save the lookup data mostly for record keeping purposes not so much for caching purposes. 

This twilio call incurs no extra cost since we aren't asking for any special info that we'd be charged for. (eg sim pumping or line type intellegence) So I think it's fine to not cache the data at this point in time, especially since we don't even know if the user will be able to confirm the number yet.